streams/src/test/java/org/apache/kafka/test/InternalMockProcessorContext.java

```diff
@@ -349,6 +353,27 @@ public Headers headers() {
         return recordContext.headers();
     }
 
+    @Override
+    public TaskType taskType() {
+        return TaskType.ACTIVE;
+    }
+
+    @Override
+    public void logChange(final String storeName,
+                          final Bytes key,
+                          final byte[] value,
+                          final long timestamp) {
+        recordCollector().send(
```

Comment:
When we merge the InternalMock with MockInternal would this be okay? cc @cadonna

Reply:
I assumed this would be fine since it's pretty much what happened before (ie users of the context would get the record collector and then call send) but I'd like to get this confirmed

Reply:
Both, `MockInternalProcessorContext` and `InternalMockProcessorContext` currently implement `RecordCollector.Supplier` (i.e., `recordCollector()`). Of course, once rebased the consolidated mock needs to implement `taskType()` and `logChange()`.
See also my comment above regarding `KEY_SERIALIZER` and `VALUE_SERIALIZER`.
@guozhangwang did you have anything specific in mind that I did not cover here?

Reply:
No I do not, just wanting to make sure we do not have any major conflicts when rebasing the other.